pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -320,60 +328,46 @@ func (r *ReconcileIBMBlockCSI) getAccessorAndFinalizerName(instance *ibmblockcsi
 func (r *ReconcileIBMBlockCSI) updateStatus(instance *ibmblockcsi.IBMBlockCSI, originalStatus csiv1.IBMBlockCSIStatus) error {
 	logger := log.WithName("updateStatus")
 	controllerPod := &corev1.Pod{}
-	controllerRestart := false
-	nodeRolloutRestart := false
-
-	controllerStatefulset := &appsv1.StatefulSet{}
-	err := r.client.Get(context.TODO(), types.NamespacedName{
-		Name:      oconfig.GetNameForResource(oconfig.CSIController, instance.Name),
-		Namespace: instance.Namespace,
-	}, controllerStatefulset)
-
+	err, controllerStatefulset := r.getControllerK8sObject(instance)
 	if err != nil {
 		return err
 	}
 
-	node := &appsv1.DaemonSet{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{
-		Name:      oconfig.GetNameForResource(oconfig.CSINode, instance.Name),
-		Namespace: instance.Namespace,
-	}, node)
-
+	err, nodeDaemonSet := r.getNodeK8sObject(instance)
 	if err != nil {
 		return err
 	}
 
-	instance.Status.ControllerReady = controllerStatefulset.Status.ReadyReplicas == controllerStatefulset.Status.Replicas
-	instance.Status.NodeReady = node.Status.DesiredNumberScheduled == node.Status.NumberAvailable
+	if !instance.Status.ControllerReady {
+		err := r.getControllerPod(controllerStatefulset, controllerPod)
+		if err != nil {
+			logger.Error(err, "failed to get controller pod")
+			return err
+		}
+
+		for _, containerstatus := range controllerPod.Status.ContainerStatuses {
+			if containerstatus.State.Waiting != nil {
+				if containerstatus.State.Waiting.Reason == "ImagePullBackOff" {
+					logger.Info("controller requires restart",
+											"ReadyReplicas", controllerStatefulset.Status.ReadyReplicas,
+											"Replicas", controllerStatefulset.Status.Replicas)
+					r.restartControllerPod(controllerPod)
+				}
+			}
+		}
+	}
+
+	instance.Status.ControllerReady, instance.Status.NodeReady = r.getDriverPodsStatus(instance, 
```

Comment:
Why do we need to assign the values in the `instance.Status`?
The function already does that.
I see two options:
1. remove the assignment in the function
2. remove the assignment here and rename the function since it doesn't get anything.